rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -2019,8 +2034,171 @@ extern void ppu_initialize()
 		ppu_initialize(*ptr);
 	}
 
-	// Initialize SPU cache
-	spu_cache::initialize();
+	const auto thread = g_fxo->init<named_thread>("SPRX Loader"sv, [is_gui, prx_list = std::move(prx_list)]
+	{
+		const std::string firmware_sprx_path = g_cfg.vfs.get_dev_flash() + "sys/external/";
+
+		std::vector<std::string> dir_queue;
+		dir_queue.emplace_back(Emu.GetBoot() + '/');
+		dir_queue.emplace_back(g_cfg.vfs.get_dev_flash() + "sys/external/");
```

Comment:
Why you scan dev_flash unconditionally?

Reply:
Also it seems identical to `firmware_sprx_path` variable.

Reply:
I mentioned it in https://github.com/RPCS3/rpcs3/pull/9664#issuecomment-768990072

Reply:
Also it's for main_window::OnMissingFw() to not call CreateFirmwareCache() which requires terminating current process. (here it's in ready state)
So cache creation has been moved to here instead.

Reply:
Anyway, rescanning everything can become very slow even in the case everything is compiled. Need to trade some reliability to the ability to boot at least as fast as before. In other words, find some way to determine the necessity very fast.